### PR TITLE
Store commands autocomplete

### PR DESCRIPTION
**Pull request by facundobatista**
This depends on #93

### CODE REVIEW

File: completion.bash [hunk 1]
```diff
@@ -42,10 +42,28 @@ _charmcraft()
             COMPREPLY=( $(compgen -W "--help" -- "$cur") )
             return
             ;;
+
+        # store related commands
```
Comment (chipaca) [inline completion.bash:46]: misplaced comment is misplaced

File: completion.bash [hunk 2]
```diff
@@ -78,6 +96,13 @@ _charmcraft()
                     ;;
             esac
             ;;
+        "upload")
+            case "$prev" in
+                --charm-file)
+                    _filedir charm
+                    ;;
+            esac
+            ;;
```
Comment (chipaca) [inline completion.bash:105]: also `release`'s `--revision` should autocomplete, n'eh?

File: completion.bash [hunk 3]
```diff
@@ -42,10 +42,28 @@ _charmcraft()
             COMPREPLY=( $(compgen -W "--help" -- "$cur") )
```
Comment (chipaca) [inline completion.bash:42]: way up there ^^^ there's a `cmds=(...)` that lists all the commands the completer knows about...

File: completion.bash (hunk 1 above)
Comment (facundobatista) [reply completion.bash:46]: indeed, fixed, thanks!

File: completion.bash (hunk 2 above)
Comment (facundobatista) [reply completion.bash:105]: As we discussed, let's not hit the Store to autocomplete.

File: completion.bash (hunk 1 above)
Comment (facundobatista) [reply completion.bash:46]: FTR, I removed the "message as a separation/grouping", as I like to keep more simple (grouping everything that doesn't have anything else than --help)

File: completion.bash (hunk 3 above)
Comment (facundobatista) [reply completion.bash:42]: Ah, nice, thanks

**Comment by chipaca** [comment]
Here's the [delta] FWIW

[delta]: https://github.com/facundobatista/charmcraft/compare/store-status...facundobatista:store-commands-autocomplete

File: completion.bash [hunk 4]
```diff
@@ -19,7 +19,7 @@
 _charmcraft()
 {
     local cur prev words cword cmd cmds
-    cmds=(build version login logout whoami)
+    cmds=(build version login logout whoami names upload revisions status release)
```
Comment (jameinel) [inline completion.bash:22]: this is starting to get big enough that it feels like something we'd want to start automatically generating.
Manual is still fine for now, but it does mean having to keep 2 things up to date whenever adding an argument or optional parameter.

**Pull request merged**